src/go/k8s/pkg/console/sasl.go

```diff
@@ -184,27 +184,16 @@ func NewKafkaACL(
 
 // Ensure implements Resource interface
 func (k *KafkaACL) Ensure(ctx context.Context) error {
-	// Build ACL for console SASL user to access everything
-	b := kadm.NewACLs().
-		Allow(GenerateSASLUsername(k.consoleobj)).
-		Topics("*").Groups("*").Clusters().Operations(kadm.OpAll).
-		ResourcePatternType(kadm.ACLPatternLiteral)
-	if err := b.ValidateCreate(); err != nil {
-		return fmt.Errorf("validating create ACLs: %w", err)
-	}
-	b.PrefixUserExcept()
-
-	kadmclient, err := k.kafkaAdmin(ctx, k.Client, k.clusterobj, k.store)
+	kadmclient, b, err := k.createAdminClient(ctx)
 	if err != nil {
-		return fmt.Errorf("creating kafka admin client: %w", err)
+		return fmt.Errorf("creating console sasl user: %w", err)
 	}
 
 	results, err := kadmclient.CreateACLs(ctx, b)
+	var errList []error
 	if err != nil {
-		return fmt.Errorf("creating kafka ACLs: %w", err)
+		errList = append(errList, err)
```

Comment:
Why append and not return immediately?

Reply:
because you end up having error with the same context `creating kafka ACLs: %w`. If you would see logs of the operator you wouldn't know which one error out.

https://github.com/redpanda-data/redpanda/blob/74a9711b8e783d5afacc9b52374908b9aa964191/src/go/k8s/pkg/console/sasl.go#L204
https://github.com/redpanda-data/redpanda/blob/74a9711b8e783d5afacc9b52374908b9aa964191/src/go/k8s/pkg/console/sasl.go#L214

Reply:
Hmm I think in any case it is an error in creating ACL and the wrapped one will give more info. Also in this case we still won't know which one error out because we append.

Reply:
We don't know in either case. The context that is put in front of the error doesn't differentiate the error. I didn't want to spend time on arguing what context it should be. I was lazy in that case and combined all errors.